examples/dev/leaking_check.py

```diff
@@ -0,0 +1,27 @@
+import os
+import psutil
+import weakref
+import gc
+import objgraph
+
+import napari
+import numpy as np
+
+process = psutil.Process(os.getpid())
+viewer = napari.Viewer()
+
+print("mem", process.memory_info().rss)
+
+for i in range(15):
+    print(viewer.add_image(np.random.random((60, 1000, 1000))).name)
+
+print("mem", process.memory_info().rss)
+
+napari.run()
+
+li = weakref.ref(viewer.layers[0])
+del viewer.layers[0]
+
+objgraph.show_backrefs(li(), filename="test.png")
```

Comment:
this produce nice image with reference information like this:
![test](https://user-images.githubusercontent.com/3826210/130441494-57e1c76b-1698-47f7-9c1b-75c2e4f9c048.png)

Reply:
cool!